### PR TITLE
Bb chapter 8, 2nd ed.

### PR DESCRIPTION
**Pull request by mb706**

### CODE REVIEW

**Comment by mb706** [comment]
I am done here

File: book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd [hunk 1]
```diff
@@ -561,6 +561,17 @@ Tuning the `selection` hyperparameters can help determine which of the possible
...
+# instead of plotting, we make autoplot() save the plot so we can edit it afterwards
+# This is *not* the same as ggplot::last_plot(), but the result is easier to handle in a loop.
+plt_container = new.env()
+autoplot = function(...) {
+  # <<- doesn't seem to work on CI for some reason
+  plt_container$plt = ggplot2::autoplot(...)
+  invisible(NULL)
+}
+```
+
```
Comment (RaphaelS1) [inline book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd:574]: @mb706 this does not work as intended. Please check in the pdf output but this is quite broken...

Screenshot 2023-07-03 at 08 44 04

Also figures I previously fixed are now printing badly again, e.g. this is now a full page

Screenshot 2023-07-03 at 08 44 31

File: book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd [hunk 2]
```diff
@@ -572,12 +583,44 @@ graph_learner$param_set$set_values(
...
+  et = element_text(angle = 45)
+  plt[[i]]$theme$axis.text.x = et
+  ### Remove the legends
+  plt[[i]]$theme$legend.position = "none"
+}
+autoplot = ggplot2::autoplot # reset to original function
+print(plt)
+```
+
+As we can see in the plot, the KNN-learner with `k` set to `r instance$result$classif.kknn.k` was selected, which performs best in combination with the Yeo-Johnson transform.
```
Comment (RaphaelS1) [inline book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd:623]:
```suggestion
As we can see, the KNN-learner with `k` set to `r instance$result$classif.kknn.k` was selected, which performs best in combination with the Yeo-Johnson transform.
```

**Review by RaphaelS1 — CHANGES_REQUESTED** [review]

File: book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd (hunk 1 above)
Comment (mb706) [reply book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd:574]: I am fixing the warning message, and apparently the autoplot() got lost when merging master. but are you sure the plot you are showing is not an artifact of how you are rendering a single file, or broken cache? This one looks fine for me

Comment (mb706) [reply book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd:574]: (or maybe an old mlr3pipelines version?)

**Comment by mb706** [comment]
I added some arguments to suppress warnings, but I notice that plotting breaks when the cache is in a broken state. If the result does not look like this, you can try deleting the cache
image

**Pull request merged**